misc-utils/exch.c

```diff
@@ -26,14 +26,22 @@
 # define RENAME_EXCHANGE (1 << 1)
 #endif
 
-#if !defined(HAVE_RENAMEAT2) && defined(SYS_renameat2)
-static inline int renameat2(int olddirfd, const char *oldpath,
-			    int newdirfd, const char *newpath, unsigned int flags)
-{
-	return syscall (SYS_renameat2, olddirfd, oldpath, newdirfd, newpath, flags);
-}
+static inline int rename_exchange(int dirfd,
+				  const char *oldpath, const char *newpath) {
+	int rc;
+
+#if defined(HAVE_RENAMEAT2)
+	rc = renameat2(dirfd, oldpath, dirfd, newpath, RENAME_EXCHANGE);
+#elif defined(SYS_renameat2)
+	rc = syscall (SYS_renameat2,
+		      dirfd, oldpath, dirfd, newpath, RENAME_EXCHANGE);
+#else
+	rc = -1;
```

Comment:
What do you think setting `ENOSYS` to `errno` in addition to setting -1 to `rc`.
So the next commit that you proposed in this pull request can be removed.

I'm not sure what I wrote here is a good way.
@karelzak @t-8ch may give us hints.

Reply:
Ah, that success came from errno.

errno = ENOSYS; gives me this:
```
exch: failed to exchange "a" and "b": Function not implemented
```

which is indeed better.